utility/repoutil.py

```diff
@@ -315,6 +202,8 @@ class PullRequestCreationDetail(t.TypedDict, total=False):
     maintainer_can_modify: bool
     draft: bool
     issue: int
+    head: str
+    base: str
```

Comment:
these fields are set using the positional arguments ```head_branch``` and ```base_branch``` in the ```create_pull_request``` function itself

Reply:
I understand that there can be a warning, but it's not that critical in my opinion, adding these args to the ```PullRequestCreationDetail``` isn't good, because we can't be sure that these args are actually provided, and raising an error here will not really help us, as the solution I would prefer doing something like: 
```
request_body = request_body or PullRequestCreationDetail()
request_body = request_body.copy()
request_body['head'] = head_branch
request_body['base'] = base_branch
```
as far as I understand, it shouldn't highlight any warnings

Reply:
I agree that it's not critical in this specific case, but the type checker does catch important errors and it's easier to spot them when there aren't other unimportant warnings in the file.

I found an approach that works gets rid of the error.
```
request_body_dict = dict(request_body or PullRequestCreationDetail())
request_body_dict['head'] = head_branch
request_body_dict['base'] = base_branch
```